src/Parser/DeathKnight/Unholy/Modules/Features/DarkTransformationAndWounds.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+import Wrapper from 'common/Wrapper';
+
+class DarkTransformationAndWounds extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  // used to track how many stacks a target has
+  targets = {};
+
+  darkTransformationActive = 0;
+  darkTransformationCasts = 0;
+  stopWatchActive = 0;
+  stopWatch = 0;
+  totalTime = 0;
+  
+  
+  on_byPlayer_applybuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = 1;
+		  this.darkTransformationCasts += 1;
+	  }
+  }
+  
+  on_byPlayer_removebuff(event){
+	  const spellId = event.ability.guid;
+	  if(spellId === SPELLS.DARK_TRANSFORMATION.id){
+		  this.darkTransformationActive = 0;
+	  }
+  }
+  
+  on_byPlayer_applydebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+	  }
+  }
+
+  on_byPlayer_removedebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+    }
+  }
+
+  on_byPlayer_removedebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		  this.targets[encodeTargetString(event.targetID, event.targetInstance)] = 0;
+	  }
+  }
+
+  on_byPlayer_cast(event){
+    if(this.targets.hasOwnProperty(encodeTargetString(event.targetID, event.targetInstance))) {
+	    if (this.targets[encodeTargetString(event.targetID, event.targetInstance)] === 0) {
+		    if (this.darkTransformationActive === 1) {
+			    if (this.stopWatchActive === 0) {
+				    this.stopWatch = event.timestamp;
+				    this.stopWatchActive = 1;
+			    }
+		    } 
+		    else if (this.stopWatchActive === 1) {
```

Comment:
You can avoid some of the duplicate code and the nested if statements if you do this check first thing in `on_byPlayer_cast`, assuming you always wants to add the time if the stopwatch is active. Then you can stop/start it depending on whether the buff or debuffs are up afterwards.

Reply:
I'm not too sure I can do that. I only add the time when the conditions no longer hold, so only when the stopwatch turns from active to inactive.

Reply:
I was thinking you could just start it again afterwards. Anyway I guess this works, it's only two extra lines after all.